polaris-core/src/main/java/org/apache/polaris/core/storage/PolarisStorageConfigurationInfo.java

```diff
@@ -160,7 +161,7 @@ public static Optional<PolarisStorageConfigurationInfo> forEntityPath(
                 Set<String> locations =
                     StorageUtil.getLocationsAllowedToBeAccessed(
                         null, entityPathReversed.get(0).getPropertiesAsMap());
-                return new StorageConfigurationOverride(
+                return new LocationRestrictions(
```

Comment:
I think the TODO on line 160 added in https://github.com/apache/polaris/pull/1320/files#diff-607cdd8c6ea78443988359e42ffec3091a4c84bc9151575772ee3b90164e03f3 was correctly pointing out that the addition of these locations from `StorageUtil.getLocationsAllowedToBeAccessed` (formerly `userSpecifiedWriteLocations`) msimatched the intent of the validation in ways that were problematic for Views.

There also appears to have been an unintended change in semantics in https://github.com/apache/polaris/pull/2149 that started adding *all* table-requested paths instead of just the write.data.path and write.metadata.path.

We should remove all this location-augmentation here and simply return the basic restrictions directly from `configInfo` in this `else` branch.

Reply:
I don't think we need to worry too much about edge cases like a custom `write.metadata.path` for views. In fact, even for tables, we all know that tables can contain data files outside of `write.metadata.path` U `write.data.path` U `location` right? We need to capture the majority of use layouts with credential vending, but not necessarily the long tail.

Reply:
Right, regardless of whether this allows or disallows setting `write.metadata.path` for Views, that test case is currently documenting the incorrect behavior that also applies to Tables.

Probably better to fix-forward the test; even #1320 itself indicated it was just documenting current observed behavior, but we can update the `createViewWithCustomMetadataLocationUsingPolaris` to ensure it also fails with ForbiddenException even if the custom path is in the parent.

In this case we also want to make sure even with `ALLOW_UNSTRUCTURED_TABLE_LOCATION=true` for the *CreateTable* scenario (not CreateView), the creation will fail if the "custom location" isn't part of the Catalog's `allowedLocations`.

Reply:
Note, if we want a "success" case for Views calling `super.createViewWithCustomMetadataLocation();` we can just construct the original Catalog's StorageConfig to include the custom path in `allowedLocations`

Reply:
Fixed